packages/lsp-extension/src/index.ts

```diff
@@ -0,0 +1,274 @@
+    const languageServerSettings = (options.language_servers ||
+      {}) as TLanguageServerConfigurations;
+
+    connectionManager.initialConfigurations = languageServerSettings;
+    // TODO: if priorities changed reset connections
+    connectionManager.updateConfiguration(languageServerSettings);
+    connectionManager.updateServerConfigurations(languageServerSettings);
+    connectionManager.updateLogging(
+      options.logAllCommunication,
+      options.setTrace
+    );
+  };
+  languageServerManager.sessionsChanged.connect(() => {
+    settingRegistry.transform(plugin.id, {
+      fetch: plugin => {
+        const schema = plugin.schema.properties!;
+        const defaultValue: { [key: string]: any } = {};
+        languageServerManager.sessions.forEach((_, key) => {
+          defaultValue[key] = { priority: 50, serverSettings: {} };
+        });
+
+        schema[LANGUAGE_SERVERS]['default'] = defaultValue;
+        return plugin;
+      },
+      compose: plugin => {
+        const properties = plugin.schema.properties!;
+        const user = plugin.data.user;
+
+        const serverDefaultSettings = properties[LANGUAGE_SERVERS][
+          'default'
+        ] as PartialJSONObject;
+        const serverUserSettings = user[LANGUAGE_SERVERS] as
+          | PartialJSONObject
+          | undefined;
+        let serverComposite = { ...serverDefaultSettings };
+        if (serverUserSettings) {
+          serverComposite = { ...serverComposite, ...serverUserSettings };
+        }
+        const composite: { [key: string]: any } = {
+          [LANGUAGE_SERVERS]: serverComposite
+        };
+        Object.entries(properties).forEach(([key, value]) => {
+          if (key !== LANGUAGE_SERVERS) {
+            if (key in user) {
+              composite[key] = user[key];
+            } else {
+              composite[key] = value.default;
+            }
+          }
+        });
+        plugin.data.composite = composite;
+        return plugin;
+      }
+    });
+    settingRegistry
+      .load(plugin.id)
+      .then(settings => {
+        updateOptions(settings);
+        settings.changed.connect(() => {
+          updateOptions(settings);
+        });
+      })
+      .catch((reason: Error) => {
+        console.error(reason.message);
+      });
+  });
+
+  // Add a sessions manager if the running extension is available
+  if (runningSessionManagers) {
+    addRunningSessionManager(
+      runningSessionManagers,
+      connectionManager,
+      translator
+    );
+  }
+
+  if (settingRendererRegistry) {
+    settingRendererRegistry.addRenderer(
+      LANGUAGE_SERVERS,
+      (props: FieldProps) => {
+        return renderServerSetting(props, translator);
+      }
+    );
+  }
+
+  return connectionManager;
+}
+
+function activateFeature(
+  app: JupyterFrontEnd,
+  settingRegistry: ISettingRegistry,
+  translator: ITranslator
+): ILSPFeatureManager {
+  const featureManager = new FeatureManager();
+
+  return featureManager;
+}
```

Comment:
Maybe this one can be inlined at the plugin definition directly if it just `return new FeatureManager();`?